pkg/component/worker/containerd/runtime.go

```diff
@@ -0,0 +1,178 @@
+        container_annotations = []
+        pod_annotations = []
+        privileged_without_host_devices = false
+        runtime_engine = ""
+        runtime_path = ""
+        runtime_root = ""
+        runtime_type = ""
+
+        [plugins."io.containerd.grpc.v1.cri".containerd.default_runtime.options]
+
+      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
+
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+          base_runtime_spec = ""
+          cni_conf_dir = ""
+          cni_max_conf_num = 0
+          container_annotations = []
+          pod_annotations = []
+          privileged_without_host_devices = false
+          runtime_engine = ""
+          runtime_path = ""
+          runtime_root = ""
+          runtime_type = "io.containerd.runc.v2"
+
+          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+            BinaryName = ""
+            CriuImagePath = ""
+            CriuPath = ""
+            CriuWorkPath = ""
+            IoGid = 0
+            IoUid = 0
+            NoNewKeyring = false
+            NoPivotRoot = false
+            Root = ""
+            ShimCgroup = ""
+            SystemdCgroup = false
+`
+
+const importsPath = "/etc/k0s/containerd.d/*.toml"
+const containerdCRIConfigPath = "/run/k0s/containerd-cri.toml"
+
+type Config struct {
+	loadPath string
+
+	log *logrus.Entry
+}
+
+func NewConfig() *Config {
+
+	return &Config{
+		loadPath: importsPath,
+		log:      logrus.WithField("component", "containerd"),
+	}
+}
+
+// Resolves imports from the import glob path
+// If the partial config has CRI plugin enabled, it will add the runc CRI config (single file)
+// if no CRI plugin is found, it will add the file as-is to imports
+func (c *Config) ResolveImports() ([]string, error) {
+	var imports []string
+	var criConfigBuffer bytes.Buffer
+	_, err := criConfigBuffer.WriteString(runcCRIConfig)
+	if err != nil {
+		return nil, err
+	}
+	files, err := filepath.Glob(c.loadPath)
+	c.log.Debugf("found containerd config files: %v", files)
+	if err != nil {
+		return nil, err
+	}
+	for _, file := range files {
+		data, err := os.ReadFile(file)
+		if err != nil {
+			return nil, err
+		}
+		c.log.Debugf("parsing containerd config file: %s", file)
+		hasCRI, err := c.hasCRIPluginConfig(data)
+		if err != nil {
+			return nil, err
+		}
+		if hasCRI {
+			c.log.Debugf("found CRI plugin config in %s, appending to CRI config", file)
+			// Append all CRI plugin configs into single file
+			// and add it to imports
+			criConfigBuffer.WriteString(fmt.Sprintf("# appended from %s\n", file))
+			_, err := criConfigBuffer.Write(data)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			c.log.Debugf("adding %s as-is to imports", file)
+			// Add file to imports
+			imports = append(imports, file)
+		}
+	}
+
+	// Write the CRI config to a file and add it to imports
+	err = os.WriteFile(containerdCRIConfigPath, criConfigBuffer.Bytes(), 0644)
```

Comment:
Looks a bit fishy to me. By the name of the function, I wouldn't expect a file writing here. Not sure, how to improve it tho, so we can leave it as is for now.